pkg/blockbuilder/tsdb.go

```diff
@@ -44,6 +44,23 @@ type TSDBBuilder struct {
 	tsdbs   map[tsdbTenant]*userTSDB
 }
 
+var softErrProcessor = mimir_storage.SoftAppendErrorProcessor{
```

Comment:
Now that we aren't checking for nils, this initialization would be good to tuck into a `NewSoftError...` initializer so that later when someone adds a new error type it will be a compile error and not a lurking panic because they didn't add callbacks to all of the places this struct gets initialized.